argilla-server/pyproject.toml

```diff
@@ -59,6 +59,7 @@ dependencies = [
     "typer >= 0.6.0, < 0.10.0", # spaCy only supports typer<0.10.0
     "packaging>=23.2",
     "psycopg2-binary>=2.9.9",
+    "datasets>=3.0.1",
```

Comment:
I think this is a bit strict. That said, it's the server so compatibility is less important.